apps/src/aichat/views/teacherFeedback/CleanFeedbackFooter.tsx

```diff
@@ -0,0 +1,80 @@
+import classNames from 'classnames';
+import React, {memo} from 'react';
+
+import {Role} from '@cdo/apps/aiComponentLibrary/chatMessage/types';
+import {Button, buttonColors} from '@cdo/apps/componentLibrary/button';
+import {WithTooltip} from '@cdo/apps/componentLibrary/tooltip';
+import {EmText} from '@cdo/apps/componentLibrary/typography';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+import {AiChatTeacherFeedback as TeacherFeedback} from '@cdo/generated-scripts/sharedConstants';
+
+import aichatI18n from '../../locale';
+import {submitTeacherFeedback} from '../../redux/aichatRedux';
+import {FeedbackValue} from '../../types';
+
+import moduleStyles from './teacher-feedback-footer.module.scss';
+
+interface Props {
+  id: number;
+  teacherFeedback?: FeedbackValue;
+  role: Role;
+}
+
+/**
+ * Teacher feedback footer displayed for messages without any profanity violations or errors.
+ */
+const CleanFeedbackFooter: React.FC<Props> = ({id, teacherFeedback, role}) => {
```

Comment:
The code for these components is pretty similar to before, I just felt it would be nicer to split them out into separate files. The main notable changes are that the state has been removed, and tooltips have been added.

Reply:
Was thinking the same thing in your previous PR, I like it!